src/OpenIddict.Core/Infrastructure/OpenIddictProvider.Serialization.cs

```diff
@@ -17,14 +17,41 @@
 namespace OpenIddict.Infrastructure {
     public partial class OpenIddictProvider<TUser, TApplication, TAuthorization, TScope, TToken> : OpenIdConnectServerProvider
         where TUser : class where TApplication : class where TAuthorization : class where TScope : class where TToken : class {
-        public override async Task SerializeRefreshToken([NotNull] SerializeRefreshTokenContext context) {
+        public override async Task SerializeAuthorizationCode([NotNull] SerializeAuthorizationCodeContext context) {
             var services = context.HttpContext.RequestServices.GetRequiredService<OpenIddictServices<TUser, TApplication, TAuthorization, TScope, TToken>>();
 
-            Debug.Assert(context.Request.RequestType == OpenIdConnectRequestType.TokenRequest,
-                "The request should be a token request.");
+            Debug.Assert(context.Request.RequestType == OpenIdConnectRequestType.Authentication, "The request should be an authorization request.");
+            Debug.Assert(!string.IsNullOrEmpty(context.Request.ClientId), "The client_id parameter shouldn't be null or empty.");
+
+            // Note: a null value could be returned by FindByIdAsync. In this case, throw an exception to abort the token request.
+            var user = await services.Users.FindByIdAsync(context.Ticket.Principal.GetClaim(ClaimTypes.NameIdentifier));
+            if (user == null) {
+                throw new InvalidOperationException("The token request was aborted because the user associated " +
```

Comment:
I didn't think to log the exception but rather the cause of the exception being thrown to have a more meaningful error message (as we already do in other parts)